### PR TITLE
Fix table metadata

### PR DESCRIPTION
**Pull request by aptkingston**
## Description
This PR updates tables to be better aware of grid metadata. That's things like column visibility and order, which are configured in grids in the data section.

This PR makes a number of changes to make tables behave more like grids:
- By default, tables will not show hidden columns
- By default, tables will sort columns the same way grids do (using the order from the data section). This includes showing the primary display column as the first column.
- When specifying a custom column configuration for a table, visibility and order are both overridden by this custom config
- Columns widths are specifically *not* used from the grid, as tables render columns differently and cannot be resized. Therefore they will continue to stretch columns by default to use the full width of the table.

This also fixes an issue where custom column config for tables would be completely ignored if your table contained a hidden field.

Addresses: 
- https://linear.app/budibase/issue/BUDI-7496/update-table-to-use-grid-order-metadata
- https://github.com/Budibase/budibase/issues/11788

## Feature branch env
[Feature Branch Link](http://fb-fix-table-metadata.fb.qa.budibase.net)

### CODE REVIEW

**Review by Ghrehh — APPROVED** [review]
lgtm

**Comment by deanhannigan** [comment]
- Table/Tableblock not displaying any columns by default.
  - Seems to be limited to existing tables without any visibility config specified.
  - If the column visibilty is toggled in the data in the data section, the `All columns` behaviour works as expected.

File: packages/client/src/components/app/table/Table.svelte [hunk 1]
```diff
@@ -55,32 +56,52 @@
...
     }
 
     // Otherwise generate columns
     let columns = []
     let autoColumns = []
     Object.entries(schema).forEach(([field, fieldSchema]) => {
+      if (!fieldSchema.visible) {
+        return
+      }
       if (!fieldSchema?.autocolumn) {
```
Comment (deanhannigan) [inline packages/client/src/components/app/table/Table.svelte:76]: Would this need to default to true when not set to ensure its visible?

Comment (aptkingston) [reply packages/client/src/components/app/table/Table.svelte:76]: Nice one - I think I actually meant to check against an explicit `false`, which would mean nullish is considered visible.

Comment (aptkingston) [reply packages/client/src/components/app/table/Table.svelte:76]: Pushed that up now - would you mind checking if that now works as expected?

Comment (deanhannigan) [reply packages/client/src/components/app/table/Table.svelte:76]: Reviewed this again, all default visibility checks for the Table/Tableblock working as expected 👍

**Comment by MihailHadzhiev2022** [comment]
LGTM!

**Pull request merged**